core/shared/src/main/scala/zio/logging/internal/json/SmallBooleanStack.scala

```diff
@@ -0,0 +1,35 @@
+package zio.logging.internal.json
+
+final class SmallBooleanStack(
+  private val data: Long
+) extends AnyVal {
+  /* 64 - stackSize reserved for index store -> 6 bits are enough to represent 58 states */
+  @inline final private def stackSize      = 58
+  @inline final private def idxIncrement   = 0x400000000000000L // 1L << stackSize
+  @inline final private def undefinedValue = 0xffffffffffffffffL
+
+  @inline final private def stateIdx      = data >>> stackSize
+  @inline final private def increaseIdx() = new SmallBooleanStack(data + idxIncrement)
+
+  @inline final private def setState(state: Boolean): Long =
+    if (state) data | (1L << stateIdx)
+    else data & ~(1L << stateIdx)
+
+  def isDefined()      = data != undefinedValue
+  def isEmpty: Boolean = stateIdx == 0
+
+  def pop(): SmallBooleanStack = {
+    if (stateIdx == 0) new SmallBooleanStack(undefinedValue)
+    new SmallBooleanStack(data - idxIncrement)
+  }
+
+  def push(value: Boolean): SmallBooleanStack =
+    if (stateIdx >= stackSize - 1) new SmallBooleanStack(undefinedValue)
+    else new SmallBooleanStack(setState(value)).increaseIdx()
```

Comment:
this branch creating SmallBooleanStack twice (second time in increaseIdx)

Reply:
Not relevant anymore.